### ISSUE TITLE
This appears to operate token-by-token?

### ISSUE THREAD

**Issue by domenic** [ricea/encoding-streams#9]
I'd always envisioned that the chunks would be "roughly" the same size on both the readable and writable sides, with perhaps some trailing bytes left over when decoding. This spec, however, seems to say that the readable side's chunks will always be created by processing a single token, and enqueuing each.

Is this intended? Are we worried about this behavior? Is it better, or worse, for implementers?

**Comment by ricea** [comment]
Nope, it's just confusing. The controlling steps are

"If token is end-of-stream, run these substeps:" for decode, and "If result is finished, run these substeps:" for encode. In both cases, this means `enqueue()` isn't called until all the input has been processed.

The resulting behaviour is that for decoding, the output chunk contains every complete code point from the input (incomplete code points will appear in a subsequent chunk, or result in error behaviour if the stream is then closed). For encoding, the entire input string is output every time.

IIRC, I already wrote test(s) for this case: https://github.com/GoogleChromeLabs/text-encode-transform-prollyfill/tree/master/tests

It might be slightly less confusing if/when we rename the Encoding Standard's concept of stream to "token sequence". But I think it'll always require a careful reading of the state machine to make sense of it.

Closing in favour of #1.
